### PR TITLE
[terra-badge] Update docs to use MDX syntax

### PR DESCRIPTION
**Pull request by pranav300**
### Summary
Updated docs to use MDX syntax.

### CODE REVIEW

File: packages/terra-badge/src/terra-dev-site/doc/badge/Badge.1.doc.mdx [hunk 1]
```diff
@@ -1,3 +1,14 @@
+import { Badge } from 'terra-badge/package.json?dev-site-package';
+
+import BadgeIntent from '../example/BadgeIntent?dev-site-example';
+import BadgeSize from '../example/BadgeSize?dev-site-example';
+import BadgeIcon from '../example/BadgeIcon?dev-site-example';
+import BadgeVisuallyHiddenText from '../example/BadgeVisuallyHiddenText?dev-site-example';
+
+import BadgePropsTable from '../../../../src/Badge?dev-site-props-table';
```
Comment (mjhenkes) [inline packages/terra-badge/src/terra-dev-site/doc/badge/Badge.1.doc.mdx]:
```suggestion
import BadgePropsTable from 'terra-badge/src/Badge?dev-site-props-table';
```

File: packages/terra-badge/src/terra-dev-site/doc/badge/ChangeLog.3.doc.mdx [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { Badge } from 'terra-badge/package.json?dev-site-package';
+
+import ChangeLog from '../../../../CHANGELOG.md';
```
Comment (mjhenkes) [inline packages/terra-badge/src/terra-dev-site/doc/badge/ChangeLog.3.doc.mdx]:
```suggestion
import ChangeLog from 'terra-badge/CHANGELOG.md';
```

Comment (pranav300) [reply packages/terra-badge/src/terra-dev-site/doc/badge/ChangeLog.3.doc.mdx]: Updated here https://github.com/cerner/terra-core/pull/2811/commits/8dced66a888bcf4af6d13f356a94e8d111e6b801.

File: packages/terra-badge/src/terra-dev-site/doc/badge/Badge.1.doc.mdx (hunk 1 above)
Comment (pranav300) [reply packages/terra-badge/src/terra-dev-site/doc/badge/Badge.1.doc.mdx]: Updated here https://github.com/cerner/terra-core/pull/2811/commits/8dced66a888bcf4af6d13f356a94e8d111e6b801.

**Pull request merged**